### DISCUSSION TITLE
SUB INVOICE LINE, how to use it

### DISCUSSION THREAD

**Discussion by PhilippGoetje** [ZUGFeRD/mustangproject#616] [Q&A] [answered]
Hey,

I was asked how the Mustangproject is handling sub-positions aka sub invoice lines currently and I could not find any documentation. Can somebody confirm, if it is implemented or not? Both, the ZUGFeRD and the xRechnung know of the sub-positions in their extended specs. 
Thanks in advance!

**Comment by jstaerk** [comment]
Hi,
no they dont. In ZUGFeRD it is definitely not there at all yet (it is under active discussion as their change request issue 96) and the XRechnung extension is not defined for CII.

I would summarize the issue that they successlessly try to mirror UBL features on similar, yet slightly different CII features. My contribution to the discussion in the AWV was the suggestion to simply allow CII's ParentLineID attribute. But that's probably too simple. They positively want, and say they need, redundant item-based subtotals which they could, in CII, only have if they were using IncludedSuboridinateTradeLineItem - but that would allow only one additional level of sub invoice lines, e.g. you could have a 2.1, 2.2 and 2.3 but not a 2.3.1. As far as I know they now prepare a draft proposal to UN/CEFACT to change CII.

What do you need it for? 
Why can't you just use invoice line IDs like "2.1" and "2.2"?
And would you be willing to contribute to it's development?

**Reply by PhilippGoetje** [reply]
Hey Jochen,

thank you for your answer. 
I am not a Java developer and therefore my input will sadly be limited to ideas, discussions and information around the intended implementation of other users as well as testing within Ninox. 
If that helps, I am willing to contribute what I can.

The "faked" system with line IDs is what I had suggested too and will discuss this further as an intermediate solution.

**Comment by msccip** [comment]
Since XRechnung version 2.2, the XR Extension is also available for CII. There is even an example invoice in the test set, but without sublines ;-)

**Reply by jstaerk** [reply]
oh cool where did you find that? where is the spec? 

https://xeinkauf.de/faq/xrechnung#technischeumsetzung still says
```
Warum gibt es zur Extension XRechnung kein Syntax-Binding für UN/CEFACT XML Industry Invoice D16B?
Das Syntax-Binding zur Syntax UN/CEFACT XML Industry Invoice D16B kann erst veröffentlicht werden, wenn auf Seiten des Syntaxbetreibers die notwendigen technischen Voraussetzungen geschaffen sind. Dies Arbeiten finden aktuell statt, Syntaxbetreiber und KoSIT stimmen sich zu einer technischen Lösung ab. Die Veröffentlichung geschieht dann im Rahmen des regulären Releasezyklus von XRechnung.
```

**Comment by msccip** [comment]
It has been in the scenarios.xml of the KoSIT validator since V2.2 and in the test suite there is the example invoice 04.05a-INVOICE_uncefact.xml. But, as mentioned, without subpositions. It is primarily about adjustments for the health insurance companies (Krankenkassen).

**Comment by jstaerk** [comment]
The reference is quite interesting but unfortunately the example contains only one item, or am I missing the subitem somehow?

**Comment by msccip** [comment]
As I already said "without subpositions" ...

**Comment by jstaerk** [comment, marked as answer]
I'm closing this discussion as it does not make sense to talk about features which are not yet standardized.

**Reply by PhilippGoetje** [reply]
Can you help me understand, why this is not standardized, when this document: https://xeinkauf.de/app/uploads/2023/02/231-XRechnung-2023-02-03.pdf, linked by https://xeinkauf.de/xrechnung/ is listing this option in chapter 16.1?
Thanks!

**Reply by jstaerk** [reply]
Because it does not explain to you, me, or anyone how to do subinvoice lines in CII
